url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/model/TaskHelper.java

```diff
@@ -0,0 +1,196 @@
+package org.swiftgantt.model;
+
+
+import org.apache.commons.lang.time.DateFormatUtils;
+
+import org.swiftgantt.Config;
+import org.swiftgantt.GanttChart;
+import org.swiftgantt.common.Constants;
+import org.swiftgantt.common.Time;
+import org.swiftgantt.ui.TimeUnit;
+
+/**
+ * 
+ * @author Yuxing Wang
+ * @since 0.2.4
+ */
+public class TaskHelper {
+
+	private TaskHelper() {
+	}
+
+	/**
+	 * Format time to display.
+	 * 
+	 * @param timeUnit
+	 * @param time
+	 * @return
+	 */
+	public static String formatTimeByTimeUnit(TimeUnit timeUnit, Time time) {
+		String retValue = null;
+		if (timeUnit == TimeUnit.Hour || timeUnit == TimeUnit.AllDay) {
+			retValue = DateFormatUtils.format(time.getTimeInMillis(), Constants.TIME_FORMAT_YEAR_HOUR_BAR);
+			//retValue = TimeFormatter.formatCalendar(time, TimeFormatter.TIME_FORMAT_YYYY_MM_DD_HH_MM_SS_BAR);
+		} else if (timeUnit == TimeUnit.Day) {
+			retValue = DateFormatUtils.format(time.getTimeInMillis(), "yyyy-MM-ddEEE");
+		} else if (timeUnit == TimeUnit.Week) {
+			retValue = DateFormatUtils.format(time.getTimeInMillis(), Constants.TIME_FORMAT_YYYY_MM_DD_BAR);
+		} else if (timeUnit == TimeUnit.Month) {
+			retValue = DateFormatUtils.format(time.getTimeInMillis(), Constants.TIME_FORMAT_YYYY_MM_BAR);
+		} else if (timeUnit == TimeUnit.Year) {
+			retValue = time.get(Time.YEAR) + "";
+		}
+		return retValue;
+	}
+
+	/**
+	 * Time offset from start time to end time.
+	 * 
+	 * @param timeUnit
+	 * @param start
+	 * @param end
+	 * @return
+	 */
+	public static int calcOffsetByTimeUnit(TimeUnit timeUnit, Time start, Time end) {
+		int offset = 0;
+		if (timeUnit == TimeUnit.Hour || timeUnit == TimeUnit.AllDay) {
+			offset = start.getHourIntervalTo(end);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;start&#39; (index: 1 | type: Time). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;timeUnit&#39;. The same parameter type is validated here for example: url8af84d296f_mrcosta_gantt_generator_tgz-pJ8-org_gantt_generator_mrcpsp_RunJ8/src/org/swiftgantt/GanttChart.java:359. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).